src/asyncapidiff.ts

```diff
@@ -0,0 +1,43 @@
+import { Output, DiffOutputItem } from './types';
+
+/**
+ * Implements functions to deal with the diff.
+ * @class
+ * @returns {AsyncAPIDiff}
+ */
+export default class AsyncAPIDiff {
+  private output: Output;
+
+  constructor(output: string) {
```

Comment:
One last thing, I hope that in the next PR you will also enable also putting as constructor parameter the `Output` object, am I correct?

Reply:
You can make something like:

```ts
constructor(output: string | Output) {
  if (typeof output === 'string') {
    // output is a stringified JSON
    this.output = JSON.parse(output);
  } else {
    this.output = output;
  }
}
```

Reply:
Not really. The input will be a `JSON.stringify`ed value of our `diff` object data. I did this so that the output object won't hold any reference to the original object and such.

Reply:
oh wait, my github's kinda lagging, so I didn't saw that update.

Reply:
I think your approach looks better. Shall I update it now?

Reply:
>  I did this so that the output object won't hold any reference to the original spec object and such.

You can make deep copy of the output if you need, you're right in this case :) As I remember the shape of the `Output` type, you can make something like:

```ts
function deepCopyOfOutput(output: Output): Output {
  return {
    changes: output.changes(item => { ...item }),
  }
}
```

> I think your approach looks better. Shall I update it now? or in the next PR?

As you want :) Can be done in the next.